File: Introduction to Data Structures: Array and String/4. Two Pointer Technique/Array Partition/ArrayPartition.py
class Solution:
    def arrayPairSum(self, nums: List[int]) -> int:
        # edge case where nums == 0 - 2
        if len(nums) == 0:
            return 0

        if len(nums) == 1 or len(nums) == 2:
            return nums[0]

        # sort -> [1, 2, 2, 5, 6, 6]
        nums.sort()

        # iterate nums and add every other element -> 1 + 2 + 6 = 9
        result = 0

        for i in range(0, len(nums), 2):
            result += nums[i]

        return result

File: Introduction to Data Structures: Array and String/4. Two Pointer Technique/Array Partition/ArrayPartition.java
class Solution {
    public int arrayPairSum(int[] nums) {
        
        // Handle edge cases
        if (nums.length == 0) {
            return 0;
        }
        
        if (nums.length == 1 || nums.length == 2) {
            return nums[0];
        }
        
        // Create a variable to store the result
        int result = 0;
        
        // Sort the array
        Arrays.sort(nums);
        
        // Iterate every other element
        for (int i = 0; i < nums.length; i += 2) {
            
            // Add element to the result
            result += nums[i];
        }
        
        // Return result
        return result;
    }
}